src/_pytest/logging.py

```diff
@@ -596,10 +596,10 @@ def pytest_collection(self):
             if self.log_cli_handler:
                 self.log_cli_handler.set_when("collection")
 
-            if self.log_file_handler is not None:
-                with catching_logs(self.log_file_handler, level=self.log_file_level):
-                    yield
-            else:
+            with catching_logs(
```

Comment:
@thisch shouldn't we keep the reference to the anonymous capture handler?

```python
if self.log_file_handler is None:
    self.log_file_handler = LogCaptureHandler()
with catching_logs(self.log_file_handler, level=self.log_file_level):
```

Reply:
AFAICS we don't have to. TBH I don't yet understand why this fix fixes the problem in #6240. I'll take a closer look in the following days.